tests/helper/helper_generic.go

```diff
@@ -197,7 +197,9 @@ func CommonBeforeEach() CommonVar {
 	commonVar.OriginalKubeconfig = os.Getenv("KUBECONFIG")
 	if NeedsCluster(CurrentSpecReport().Labels()) {
 		LocalKubeconfigSet(commonVar.ConfigDir)
-		commonVar.Project = commonVar.CliRunner.CreateAndSetRandNamespaceProject()
+		if IsAuth(CurrentSpecReport().Labels()) {
```

Comment:
Would it be possible to check the current login status (using for example `oc whoami`), so as to make the test fail right away if the test is labeled as `unauth` but the current user is already logged in?
This way, the test could just fail sooner because the preconditions are not met.
I was testing this, and noticed that `make test-integration-cluster-unauth` was taking a lot of time to finish because I was already logged into the cluster.
(But that can be handled in a separate issue/PR if you prefer (and if you think that makes sense :-)).

Reply:
Good idea, added